hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,270 @@
+These generate commands continue until `EDN_ENABLE` bit is cleared, the `BOOT_REQ_DIS` bit is set, or the EDN is reset, at which point an `uninstantiate` command is sent to destroy the instance.
+This sequence fetches the initial random values needed for the system to boot.
+Use of boot-time request mode, though simpler in operation, is only for applications which do not require FIPS-approved random values.
+Please see the [entropy_src IP documentation]({{< relref "hw/ip/entropy_src/doc" >}}) for more information on trade-offs when creating CSRNG seeds before the completion of the FIPS-required health checks.
+
+## Example Topology
+The diagram below shows an example topology where two EDN modules are used to distribute CSRNG `genbits` to peripheral modules.
+
+![EDN Example Topology Diagram](edn_top_diag.svg)
+
+## Compatibility
+None.
+
+# Theory of Operations
+
+The EDN is for distributing random number streams to hardware blocks, via peripheral ports on on the EDN.
+Each block connected to a peripheral port is referred to as an endpoint.
+
+To enable the EDN block, set the `EDN_ENABLE` bit in the {{< regref "CTRL" >}} register..
+
+## Interaction with CSRNG Application Interface Ports
+
+The CSRNG application interface implements the "function envelopes" recommended by [NIST SP 800-90A](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90Ar1.pdf) for random number generation, and these function envelopes establish certain requirements for the order of operations.
+For instance, the application interface port must receive an explicit `instantiate` command before receiving any `generate` commands.
+The sequences of commands generated by a particular EDN are either controlled by the EDN state machine or by commands forwarded from firmware through the {{< regref "SW_CMD_REQ" >}} register.
+
+Whenever commands are directly forwarded from firmware to the CSRNG through the {{< regref "SW_CMD_REQ" >}} register, firmware must poll and clear the `CMD_ACK` bit of the {{< regref "SW_CMD_STS" >}} register before sending any further commands.
+
+There are two broad modes for state machine control: auto request mode and boot-time request mode.
+
+### Boot-time Request Mode
+
+Random values are needed by peripherals almost immediately after reset, and so boot-time request mode is the default mode to simplify interactions with the boot ROM.
+
+In boot-time request mode, the command sequence is fully hardware-controlled, and no command customization is possible.
+In this mode, the EDN automatically issues a special reduced-latency `instantiate` command followed by the default `generate` commands.
+This means, for instance, that no personalization strings or additional data may be passed to the CSRNG application interface port in this mode.
+On exiting, the EDN issues an `uninstantiate` command to destroy the associated CSRNG instance.
+To simplify the boot process, this mode is enabled as the default mode on reset.
```

Comment:
Sounds good. 
Will change in the next commit to "Since boot-time request mode is enabled as the default mode on reset, a single write to the `EDN_ENABLE` bit will start the flow of entropy in this mode."  (Note "to the endpoints" will also be removed.)